From 9e22e841bcb44a7058acf73728d61a6fdc775149 Mon Sep 17 00:00:00 2001
From: Sandor Yu <Sandor.yu@nxp.com>
Date: Tue, 27 Nov 2018 15:33:22 +0800
Subject: [PATCH 5209/5242] MLK-20471-1: ARM64: dtsi: imx8qm-hdmi-in: Connect
 HDMI RX to ISI2

commit  29c3ead874ca6170de8fea641590eb9cec4d0a7b from
https://source.codeaurora.org/external/imx/linux-imx.git

ISI channel 0,1 or 2,3 could concatenate to support 4K image.
ISI channel 1,2 couldn't support such function,
otherwise image will lost line data.

Signed-off-by: Sandor Yu <Sandor.yu@nxp.com>
(cherry picked from commit 18d3cf890a41ac773aefddd2ebe285d4b0197f5d)
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts  |   15 ++++-----------
 1 file changed, 4 insertions(+), 11 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts
index 0daadcb..427ece6 100644
--- a/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8qm-mek-hdmi-in.dts
@@ -28,11 +28,7 @@
 };
 
 /* HDMI RX */
-&isi_0 {
-	status = "disabled";
-};
-
-&isi_1 {
+&isi_2 {
 	interface = <4 0 2>;  /* <Input MIPI_VCx Output>
 						Input:  0-DC0, 1-DC1, 2-MIPI CSI0, 3-MIPI CSI1, 4-HDMI, 5-MEM
 						VCx:    0-VC0, 1-VC1, 2-VC2, 3-VC3, MIPI CSI only
@@ -40,13 +36,10 @@
 	status = "okay";
 };
 
-&isi_2 {
-	status = "okay";
-	fsl,chain_buf;
-};
-
 &isi_3 {
-	status = "disabled";
+	/* For HDMI RX 4K chain buf */
+	interface = <4 0 2>;
+	status = "okay";
 };
 
 &mipi_csi_0 {
-- 
1.7.9.5

